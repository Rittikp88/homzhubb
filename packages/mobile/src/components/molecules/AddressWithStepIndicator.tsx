import React from 'react';
import { FlatList, StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Badge, Label, Text } from '@homzhub/common/src/components';
import { PropertyAddressCountry } from '@homzhub/mobile/src/components/molecules/PropertyAddressCountry';
import { TypeOfPlan } from '@homzhub/common/src/domain/models/AssetPlan';
import { ILabelColor } from '@homzhub/common/src/domain/models/LeaseTransaction';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IProps {
  steps: string[];
  currentIndex: number;
  isStepDone: boolean[];
  primaryAddress: string;
  subAddress: string;
  propertyType: string;
  icon?: string;
  onEditPress?: () => void;
  selectedPan?: string;
  onPressSteps: (index: number) => void;
  badgeStyle?: StyleProp<ViewStyle>;
  stepContainerStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
}

export const AddressWithStepIndicator = (props: IProps): React.ReactElement => {
  const {
    steps,
    currentIndex,
    isStepDone,
    icon,
    selectedPan,
    badgeStyle,
    stepContainerStyle = {},
    containerStyle = {},
    primaryAddress,
    subAddress,
    propertyType,
    onPressSteps,
    onEditPress,
  } = props;

  const { t } = useTranslation(LocaleConstants.namespacesKey.property);

  const { rental, sell, informational, blue, green, darkTint7 } = theme.colors;

  const badgeData = (): ILabelColor => {
    switch (selectedPan) {
      case TypeOfPlan.RENT:
        return {
          label: t('forRental'),
          color: rental,
        };
      case TypeOfPlan.SELL:
        return {
          label: t('forSale'),
          color: sell,
        };
      case TypeOfPlan.MANAGE:
        return {
          label: t('manage'),
          color: informational,
        };
      default:
        return {
          label: t('forRental'),
          color: rental,
        };
    }
  };

  const renderIndicator = ({ item, index }: { item: string; index: number }): React.ReactElement => {
    const isCurrentStep = currentIndex === index;
    const stepDone = isStepDone[index];
    const iconColor = isCurrentStep ? blue : stepDone ? green : darkTint7;
    const labelStyle = [styles.stepLabel, isCurrentStep && styles.currentStepLabel, stepDone && styles.doneStepLabel];
    return (
      <View style={styles.indicatorView}>
        <Icon name={icons.roundFilled} size={30} color={iconColor} onPress={(): void => onPressSteps(index)} />
        <Label type="small" textType="semiBold" style={labelStyle}>
          {item}
        </Label>
      </View>
    );
  };

  const renderSeparator = (separatorData: { highlighted: boolean; leadingItem: string }): React.ReactElement => {
    let isDone = false;
    const indicatorSteps: string[] = steps;
    indicatorSteps.forEach((item, index) => {
      if (separatorData.leadingItem === item) {
        isDone = isStepDone[index];
      }
    });
    return <View style={[styles.separator, isDone && styles.doneSeparator]} />;
  };

  const badge = badgeData();

  return (
    <View style={[styles.container, containerStyle]}>
      <View style={styles.topView}>
        <Text type="small" textType="regular">
          {propertyType}
        </Text>
        {icon && <Icon name={icon} size={23} color={blue} onPress={onEditPress} />}
        {selectedPan && <Badge title={badge.label.toUpperCase()} badgeColor={badge.color} badgeStyle={badgeStyle} />}
      </View>
      <PropertyAddressCountry
        primaryAddress={primaryAddress}
        subAddress={subAddress}
        containerStyle={styles.addressView}
      />
      <FlatList
        keyExtractor={(step): string => step}
        data={steps}
        renderItem={renderIndicator}
        horizontal
        showsHorizontalScrollIndicator={false}
        ItemSeparatorComponent={renderSeparator}
        style={[styles.listStyle, stepContainerStyle]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
    borderRadius: 4,
    backgroundColor: theme.colors.white,
  },
  addressView: {
    marginVertical: 6,
  },
  indicatorView: {
    alignItems: 'center',
  },
  topView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  separator: {
    backgroundColor: theme.colors.darkTint7,
    width: 50,
    height: 2,
    marginTop: 10,
    opacity: 0.3,
  },
  doneSeparator: {
    backgroundColor: theme.colors.green,
  },
  listStyle: {
    marginTop: 10,
    alignSelf: 'center',
  },
  stepLabel: {
    color: theme.colors.darkTint7,
  },
  currentStepLabel: {
    color: theme.colors.blue,
  },
  doneStepLabel: {
    color: theme.colors.green,
  },
});
