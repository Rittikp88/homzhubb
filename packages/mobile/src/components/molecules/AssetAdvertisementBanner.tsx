import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
import { PaginationComponent } from '@homzhub/mobile/src/components/atoms/PaginationComponent';
import { AssetAdvertisement, AssetAdvertisementResults } from '@homzhub/common/src/domain/models/AssetAdvertisement';

interface IAssetAdvertisementBannerState {
  banners: AssetAdvertisement;
  activeSlide: number;
}

export class AssetAdvertisementBanner extends React.PureComponent<{}, IAssetAdvertisementBannerState> {
  public state = {
    banners: {} as AssetAdvertisement,
    activeSlide: 0,
  };

  public componentDidMount = async (): Promise<void> => {
    await this.fetchAdvertisementBanners();
  };

  public render(): React.ReactNode {
    const { banners, activeSlide } = this.state;
    if (!banners) {
      return null;
    }
    return (
      <View style={styles.container}>
        <SnapCarousel
          carouselData={banners?.results ?? []}
          carouselItem={this.renderCarouselItem}
          itemWidth={theme.viewport.width}
          activeIndex={activeSlide}
          onSnapToItem={this.onSnapToItem}
          testID="bannerSnap"
        />
        <PaginationComponent
          dotsLength={banners?.results?.length ?? 0}
          activeSlide={activeSlide}
          containerStyle={styles.overlay}
          activeDotStyle={styles.activeDotStyle}
          inactiveDotStyle={styles.inactiveDotStyle}
        />
      </View>
    );
  }

  private renderCarouselItem = (item: AssetAdvertisementResults): React.ReactElement => {
    return (
      <Image
        source={{
          uri: item.attachment.link,
        }}
        style={styles.carouselImage}
      />
    );
  };

  public onSnapToItem = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };

  public fetchAdvertisementBanners = async (): Promise<void> => {
    const response: AssetAdvertisement = await DashboardRepository.getAdvertisements();
    this.setState({ banners: response });
  };
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 4,
    height: 180,
    overflow: 'hidden',
  },
  overlay: {
    position: 'absolute',
    flex: 0,
    alignSelf: 'center',
    bottom: 0,
    paddingVertical: 5,
  },
  carouselImage: {
    height: '100%',
    width: '100%',
  },
  activeDotStyle: {
    borderWidth: 1,
  },
  inactiveDotStyle: {
    backgroundColor: theme.colors.darkTint6,
    borderColor: theme.colors.transparent,
  },
});
