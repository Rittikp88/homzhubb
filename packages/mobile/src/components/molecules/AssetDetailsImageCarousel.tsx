import React from 'react';
import { View, StyleSheet, Image, TouchableOpacity } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { ImageVideoPagination } from '@homzhub/common/src/components';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';

interface IProps {
  images: any;
  enterFullScreen: () => void;
  updateSlide: (index: number) => void;
  activeSlide: number;
}

export class AssetDetailsImageCarousel extends React.PureComponent<IProps> {
  public render(): React.ReactElement {
    const { activeSlide, images } = this.props;
    return (
      <View style={styles.carouselContainer}>
        <SnapCarousel
          carouselData={images}
          carouselItem={this.renderCarouselItem}
          activeIndex={activeSlide}
          sliderWidth={theme.viewport.width}
          itemWidth={theme.viewport.width}
          onSnapToItem={this.onSnapToItem}
        />
        <View style={styles.overlay}>
          <ImageVideoPagination
            currentSlide={activeSlide}
            totalSlides={images.length}
            type={images[activeSlide].attachment_type}
          />
        </View>
      </View>
    );
  }

  private renderCarouselItem = (item: any): React.ReactElement => {
    const { enterFullScreen } = this.props;
    return (
      <TouchableOpacity onPress={enterFullScreen}>
        {item.attachment_type === 'IMAGE' && (
          <Image
            source={{
              uri: item.link,
            }}
            style={styles.carouselImage}
          />
        )}
        {/* TODO: Remove once the api is integrated */}
        {item.attachment_type === 'VIDEO' && <Image source={item.thumbnail_url} style={styles.carouselImage} />}
      </TouchableOpacity>
    );
  };

  public onSnapToItem = (slideNumber: number): void => {
    const { updateSlide } = this.props;
    updateSlide(slideNumber);
  };
}

const styles = StyleSheet.create({
  carouselContainer: {
    borderRadius: 4,
    overflow: 'hidden',
    backgroundColor: theme.colors.darkTint1,
  },
  overlay: {
    position: 'absolute',
    alignSelf: 'center',
    bottom: 15,
    backgroundColor: theme.colors.carouselCardOpacity,
  },
  carouselImage: {
    height: '100%',
    width: '100%',
  },
});
