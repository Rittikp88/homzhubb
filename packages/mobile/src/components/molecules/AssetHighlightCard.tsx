import React, { Component } from 'react';
import { StyleSheet, FlatList, TouchableOpacity } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { Label } from '@homzhub/common/src/components/atoms/Text';
import { SVGUri } from '@homzhub/common/src/components/atoms/Svg';
import { AssetListingSection, PaginationComponent, SnapCarousel } from '@homzhub/mobile/src/components';
import { Amenity } from '@homzhub/common/src/domain/models/Amenity';

interface IProps {
  title: string;
  data: Amenity[];
  selectedAmenity: number[];
  onAmenityPress: (id: number) => void;
}

interface IState {
  activeSlide: number;
}

class AssetHighlightCard extends Component<IProps, IState> {
  public state = {
    activeSlide: 0,
  };

  public render(): React.ReactNode {
    const { title } = this.props;
    const { activeSlide } = this.state;
    const formattedData = this.getFormattedData();

    return (
      <AssetListingSection title={title} containerStyles={styles.container}>
        <>
          <SnapCarousel
            carouselData={formattedData}
            carouselItem={this.carouselItem}
            activeIndex={activeSlide}
            onSnapToItem={this.onSnapToItem}
            containerStyle={styles.carouselContainer}
          />
          <PaginationComponent
            containerStyle={styles.pagination}
            dotsLength={formattedData.length}
            activeSlide={activeSlide}
            activeDotStyle={styles.activeDot}
            inactiveDotStyle={styles.inactiveDot}
          />
        </>
      </AssetListingSection>
    );
  }

  private renderListItem = ({ item }: { item: Amenity }): React.ReactElement => {
    const { selectedAmenity, onAmenityPress } = this.props;
    const isSelected = selectedAmenity.includes(item.id);
    return (
      <TouchableOpacity style={styles.amenityItem} onPress={(): void => onAmenityPress(item.id)}>
        <SVGUri
          uri={item.attachment.link}
          height={30}
          width={30}
          stroke={isSelected ? theme.colors.active : undefined}
          strokeWidth={isSelected ? 0.5 : undefined}
        />
        <Label type="regular" textType="regular" style={isSelected && { color: theme.colors.blue }}>
          {item.name}
        </Label>
      </TouchableOpacity>
    );
  };

  private onSnapToItem = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };

  private getFormattedData = (): Amenity[][] => {
    const { data } = this.props;
    let index;
    const arrayLength = data.length;
    const tempArray = [];
    let chunk = [];
    for (index = 0; index < arrayLength; index += 9) {
      chunk = data.slice(index, index + 9);
      tempArray.push(chunk);
    }
    return tempArray;
  };

  private carouselItem = (item: Amenity[]): React.ReactElement => {
    return (
      <FlatList
        data={item}
        numColumns={3}
        renderItem={this.renderListItem}
        contentContainerStyle={styles.listContainer}
      />
    );
  };
}

export default AssetHighlightCard;

const styles = StyleSheet.create({
  container: {
    marginBottom: 20,
  },
  carouselContainer: {
    alignSelf: 'center',
  },
  listContainer: {
    paddingTop: 16,
  },
  pagination: {
    paddingVertical: 0,
  },
  amenityItem: {
    width: (theme.viewport.width - 32) / 3,
    alignItems: 'center',
    marginBottom: 16,
  },
  activeDot: {
    borderWidth: 1,
  },
  inactiveDot: {
    backgroundColor: theme.colors.darkTint10,
    borderWidth: 0,
  },
});
