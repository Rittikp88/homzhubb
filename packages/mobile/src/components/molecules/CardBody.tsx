import React from 'react';
import { StyleSheet, View } from 'react-native';
import { WithTranslation, withTranslation } from 'react-i18next';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Badge, Button, Label, Text } from '@homzhub/common/src/components';
import { IServiceItems } from '@homzhub/common/src/domain/models/Service';

interface ICardProps {
  title: string;
  badgeTitle?: string;
  description: string;
  serviceCost: string;
  isDetailView?: boolean;
  detailedData?: IServiceItems[];
  onPressInfo?: () => void;
  onConfirm?: () => void;
}

type Props = WithTranslation & ICardProps;

const CardBody = (props: Props): React.ReactElement => {
  const { t, title, description, serviceCost, isDetailView, badgeTitle, detailedData, onPressInfo, onConfirm } = props;

  const renderDetailedData = (item: IServiceItems, index: number): React.ReactElement => {
    const isIncluded = true;
    return (
      <View style={styles.detailContent} key={index}>
        <Icon
          name={isIncluded ? icons.checkFilled : icons.close}
          color={isIncluded ? theme.colors.completed : theme.colors.disabled}
          size={22}
        />
        <Text type="small" textType="regular" style={styles.listTitle}>
          {item.title}
        </Text>
      </View>
    );
  };

  const costOfPackage = (cost: string): string => {
    const parsedCost = parseInt(cost, 10);
    switch (parsedCost) {
      case 0:
        // @ts-ignore
        return t('common:noCost');
      case -1:
        // @ts-ignore
        return t('common:requestQuote');
      default:
        return cost;
    }
  };

  return (
    <>
      <View style={styles.content}>
        {!!badgeTitle && (
          <Badge title={badgeTitle} badgeColor={theme.colors.mediumPriority} badgeStyle={styles.badgeStyle} />
        )}
        <Text type="regular" textType="semiBold" style={styles.title}>
          {title}
        </Text>
        <Label type="large" textType="regular" style={styles.description}>
          {description}
        </Label>
      </View>
      <View style={styles.subContainer}>
        <Label type="regular" textType="regular" style={styles.feeText}>
          {t('totalServiceFee')}
        </Label>
        <View style={styles.costView}>
          <Text type="large" textType="light" style={styles.feeDesc}>
            {costOfPackage(serviceCost)}
          </Text>
          {isDetailView && <Icon name={icons.info} color={theme.colors.primaryColor} size={22} onPress={onPressInfo} />}
        </View>
      </View>
      {isDetailView && detailedData && (
        <View style={styles.detailContainer}>
          {detailedData.map((item: IServiceItems, index) => {
            return renderDetailedData(item, index);
          })}
          <Button type="primary" title={t('confirmService')} containerStyle={styles.buttonStyle} onPress={onConfirm} />
        </View>
      )}
    </>
  );
};
const HOC = withTranslation(LocaleConstants.namespacesKey.property)(CardBody);
export { HOC as CardBody };

const styles = StyleSheet.create({
  content: {
    paddingHorizontal: 24,
  },
  badgeStyle: {
    marginBottom: 12,
    alignSelf: 'flex-start',
  },
  title: {
    color: theme.colors.darkTint2,
  },
  description: {
    color: theme.colors.darkTint3,
    marginTop: 6,
  },
  subContainer: {
    backgroundColor: theme.colors.background,
    marginTop: 12,
    paddingVertical: 12,
    paddingHorizontal: 24,
  },
  costView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  feeText: {
    color: theme.colors.darkTint4,
  },
  feeDesc: {
    color: theme.colors.darkTint1,
    paddingTop: 6,
  },
  listTitle: {
    color: theme.colors.darkTint2,
    marginLeft: 20,
  },
  detailContainer: {
    paddingHorizontal: 24,
  },
  detailContent: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 20,
  },
  buttonStyle: {
    marginTop: 24,
  },
});
