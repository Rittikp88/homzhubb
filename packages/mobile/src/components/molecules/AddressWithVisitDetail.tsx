import React from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { DateUtils } from '@homzhub/common/src/utils/DateUtils';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Label, Text } from '@homzhub/common/src/components/atoms/Text';
import { PropertyAddress } from '@homzhub/common/src/components/molecules/PropertyAddress';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { VisitSlot } from '@homzhub/common/src/mocks/BookVisit';

interface IProps {
  primaryAddress: string;
  subAddress: string;
  startDate: string;
  endDate: string;
  comments?: string;
  isMissedVisit?: boolean;
  isCompletedVisit?: boolean;
  isRescheduleAll?: boolean;
  isFromProperty?: boolean;
  onPressSchedule?: () => void;
  containerStyle?: StyleProp<ViewStyle>;
}

export const AddressWithVisitDetail = (props: IProps): React.ReactElement => {
  const {
    subAddress,
    primaryAddress,
    startDate,
    comments,
    isMissedVisit,
    isCompletedVisit,
    containerStyle = {},
    onPressSchedule,
    isRescheduleAll = false,
    isFromProperty = false,
  } = props;
  const { t } = useTranslation(LocaleConstants.namespacesKey.property);
  const dateTime = DateUtils.convertTimeFormat(startDate, 'YYYY-MM-DD HH');
  const time = VisitSlot.find((item) => item.from === Number(dateTime[1]));
  const textStyle = [styles.textColor, isMissedVisit && styles.missedColor];
  return (
    <View style={containerStyle}>
      {isFromProperty ? (
        <>
          {!!comments && (
            <Text type="small" textType="semiBold" style={styles.addressContainer}>
              {comments}
            </Text>
          )}
        </>
      ) : (
        <PropertyAddress
          primaryTextType="small"
          subAddress={subAddress}
          primaryAddress={primaryAddress}
          subAddressStyle={styles.subAddress}
          containerStyle={styles.addressContainer}
        />
      )}
      <Label type="regular" style={[styles.textColor, !comments && styles.detail]}>
        {t('visitDetails')}
      </Label>
      <View style={styles.detailContainer}>
        <View style={styles.content}>
          <Text type="small" textType="semiBold" style={textStyle}>
            {DateUtils.getDisplayDate(dateTime[0], 'DD MMM')}
          </Text>
          <Icon
            name={icons.roundFilled}
            color={isMissedVisit ? theme.colors.error : theme.colors.darkTint3}
            size={8}
            style={styles.iconStyle}
          />
          <Text type="small" textType="semiBold" style={textStyle}>
            {time?.formatted}
          </Text>
        </View>
        <TouchableOpacity style={styles.content} onPress={onPressSchedule}>
          <Icon name={icons.schedule} color={theme.colors.blue} size={20} />
          <Text type="small" style={styles.scheduleText}>
            {isCompletedVisit ? t('newVisit') : isRescheduleAll ? t('rescheduleAll') : t('reschedule')}
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  subAddress: {
    marginLeft: 0,
  },
  addressContainer: {
    marginVertical: 12,
  },
  textColor: {
    color: theme.colors.darkTint3,
  },
  detail: {
    marginTop: 12,
  },
  missedColor: {
    color: theme.colors.error,
  },
  detailContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconStyle: {
    marginTop: 6,
    marginHorizontal: 6,
  },
  scheduleText: {
    color: theme.colors.blue,
    marginStart: 6,
  },
});
