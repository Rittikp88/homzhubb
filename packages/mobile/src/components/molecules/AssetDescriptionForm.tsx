import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useTranslation } from 'react-i18next';
import { FormikProps, FormikValues } from 'formik';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { theme } from '@homzhub/common/src/styles/theme';
import { FormCalendar, FormDropdown, FormTextInput, Text } from '@homzhub/common/src/components';
import { FormCounter } from '@homzhub/common/src/components/molecules/FormCounter';
import { AssetDescriptionDropdownValues } from '@homzhub/common/src/domain/models/AssetDescriptionForm';

interface IProps {
  formProps: FormikProps<FormikValues>;
  dropDownOptions: AssetDescriptionDropdownValues;
}

const AssetDescriptionForm = ({ formProps, dropDownOptions }: IProps): React.ReactElement => {
  const [t] = useTranslation(LocaleConstants.namespacesKey.property);
  // Todo (Sriram: Look into this prefill issue of carpetArea)

  return (
    <>
      <Text type="small" style={styles.headingStyle}>
        {t('assetDescription:description')}
      </Text>
      <View style={styles.formContainer}>
        <View style={styles.contentView}>
          <View style={styles.subContentView}>
            <FormTextInput
              style={styles.inputFieldStyle}
              name="carpetArea"
              label={t('carpetArea')}
              maxLength={10}
              numberOfLines={1}
              inputType="number"
              placeholder={t('common:enter')}
              formProps={formProps}
            />
          </View>
          <View style={styles.flexOne}>
            <FormDropdown
              name="areaUnit"
              label={t('areaUnit')}
              options={dropDownOptions.areaUnitDropdownValues}
              placeholder={t('selectAreaUnit')}
              maxLabelLength={8}
              formProps={formProps}
            />
          </View>
        </View>
        <FormDropdown
          label={t('facingText')}
          name="facing"
          options={dropDownOptions.facing}
          placeholder={t('propertySearch:selectFacing')}
          maxLabelLength={36}
          formProps={formProps}
        />
        <FormDropdown
          label={t('flooringType')}
          name="flooringType"
          options={dropDownOptions.typeOfFlooring}
          placeholder={t('selectFlooringType')}
          maxLabelLength={36}
          formProps={formProps}
        />
        <FormCalendar
          formProps={formProps}
          label={t('yearOfConstruction')}
          name="yearOfConstruction"
          allowPastDates
          isYearView
          placeHolder={t('assetFinancial:addDatePlaceholder')}
          placeHolderStyle={styles.placeHolderStyle}
          dateStyle={styles.dateStyle}
          textType="label"
        />
        <FormCounter
          containerStyles={styles.marginTop}
          name="totalFloors"
          maxCount={1000}
          label={t('numberOfFloorsText')}
          formProps={formProps}
        />
        <FormCounter
          containerStyles={styles.marginTop}
          name="onFloorNumber"
          label={t('onFloorText')}
          formProps={formProps}
          maxCount={formProps.values.totalFloors}
        />
      </View>
    </>
  );
};

const memoizedComponent = React.memo(AssetDescriptionForm);
export { memoizedComponent as AssetDescriptionForm };

const styles = StyleSheet.create({
  headingStyle: {
    marginTop: 16,
    paddingVertical: 16,
    paddingLeft: 16,
    backgroundColor: theme.colors.moreSeparator,
  },
  formContainer: {
    backgroundColor: theme.colors.white,
    paddingHorizontal: 24,
    paddingBottom: 24,
  },
  contentView: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  subContentView: {
    flex: 1,
    marginRight: 16,
  },
  flexOne: {
    flex: 1,
  },
  marginTop: {
    marginTop: 24,
  },
  inputFieldStyle: {
    height: 40,
  },
  dateStyle: {
    marginLeft: 0,
  },
  placeHolderStyle: {
    color: theme.colors.darkTint8,
  },
});
