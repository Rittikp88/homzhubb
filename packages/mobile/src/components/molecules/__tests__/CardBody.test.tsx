import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ServicesData } from '@homzhub/common/src/mocks/ServiceData';
import { CardBody } from '@homzhub/mobile/src/components/molecules/CardBody';

describe('CardBody', () => {
  it('should match snapshot', () => {
    const props = {
      title: 'card',
      description: 'card body',
      serviceCost: 'Free',
      isDetailView: true,
      badgeTitle: 'label',
      detailedData: ServicesData[0].service_items,
    };
    const wrapper = mount(<CardBody {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
