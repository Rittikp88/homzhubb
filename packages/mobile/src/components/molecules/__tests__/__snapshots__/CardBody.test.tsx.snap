// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBody should match snapshot 1`] = `
<withI18nextTranslation(CardBody)
  badgeTitle="label"
  description="card body"
  isDetailView={true}
  serviceCost="Free"
  title="card"
>
  <CardBody
    badgeTitle="label"
    description="card body"
    i18n={Object {}}
    isDetailView={true}
    serviceCost="Free"
    t={[Function]}
    tReady={false}
    title="card"
  >
    <View
      style={
        Object {
          "paddingHorizontal": 24,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 24,
          }
        }
      >
        <Badge
          badgeColor="#FBC02D"
          badgeStyle={
            Object {
              "alignSelf": "flex-start",
              "marginBottom": 12,
            }
          }
          title="label"
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 2,
                  "paddingHorizontal": 6,
                },
                Object {
                  "alignSelf": "flex-start",
                  "marginBottom": 12,
                },
                Object {
                  "backgroundColor": "#FBC02D",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 2,
                    "paddingHorizontal": 6,
                  },
                  Object {
                    "alignSelf": "flex-start",
                    "marginBottom": 12,
                  },
                  Object {
                    "backgroundColor": "#FBC02D",
                  },
                ]
              }
            >
              <Label
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#FFFFFF",
                  }
                }
                type="regular"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 12,
                        "lineHeight": 18,
                        "textAlign": "left",
                      },
                      Object {
                        "fontFamily": "OpenSans-Regular",
                        "fontWeight": "400",
                      },
                      Object {
                        "alignSelf": "center",
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                          "lineHeight": 18,
                          "textAlign": "left",
                        },
                        Object {
                          "fontFamily": "OpenSans-Regular",
                          "fontWeight": "400",
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "#FFFFFF",
                        },
                      ]
                    }
                  >
                    label
                  </Text>
                </Text>
              </Label>
            </View>
          </View>
        </Badge>
        <Text
          style={
            Object {
              "color": "#19324A",
            }
          }
          textType="semiBold"
          type="regular"
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "lineHeight": 30,
                  "textAlign": "left",
                },
                Object {
                  "fontFamily": "OpenSans-SemiBold",
                  "fontWeight": "600",
                },
                Object {
                  "color": "#19324A",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "lineHeight": 30,
                    "textAlign": "left",
                  },
                  Object {
                    "fontFamily": "OpenSans-SemiBold",
                    "fontWeight": "600",
                  },
                  Object {
                    "color": "#19324A",
                  },
                ]
              }
            >
              card
            </Text>
          </Text>
        </Text>
        <Label
          style={
            Object {
              "color": "#33495E",
              "marginTop": 6,
            }
          }
          textType="regular"
          type="large"
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                },
                Object {
                  "fontFamily": "OpenSans-Regular",
                  "fontWeight": "400",
                },
                Object {
                  "color": "#33495E",
                  "marginTop": 6,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 14,
                    "lineHeight": 22,
                    "textAlign": "left",
                  },
                  Object {
                    "fontFamily": "OpenSans-Regular",
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#33495E",
                    "marginTop": 6,
                  },
                ]
              }
            >
              card body
            </Text>
          </Text>
        </Label>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F0F5F9",
          "marginTop": 12,
          "paddingHorizontal": 24,
          "paddingVertical": 12,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F0F5F9",
            "marginTop": 12,
            "paddingHorizontal": 24,
            "paddingVertical": 12,
          }
        }
      >
        <Label
          style={
            Object {
              "color": "#4C5F72",
            }
          }
          textType="regular"
          type="regular"
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "lineHeight": 18,
                  "textAlign": "left",
                },
                Object {
                  "fontFamily": "OpenSans-Regular",
                  "fontWeight": "400",
                },
                Object {
                  "color": "#4C5F72",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 12,
                    "lineHeight": 18,
                    "textAlign": "left",
                  },
                  Object {
                    "fontFamily": "OpenSans-Regular",
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#4C5F72",
                  },
                ]
              }
            >
              totalServiceFee
            </Text>
          </Text>
        </Label>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#001B36",
                  "paddingTop": 6,
                }
              }
              textType="light"
              type="large"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 24,
                      "lineHeight": 24,
                      "textAlign": "left",
                    },
                    Object {
                      "fontFamily": "OpenSans-Light",
                      "fontWeight": "300",
                    },
                    Object {
                      "color": "#001B36",
                      "paddingTop": 6,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 24,
                        "lineHeight": 24,
                        "textAlign": "left",
                      },
                      Object {
                        "fontFamily": "OpenSans-Light",
                        "fontWeight": "300",
                      },
                      Object {
                        "color": "#001B36",
                        "paddingTop": 6,
                      },
                    ]
                  }
                >
                  Free
                </Text>
              </Text>
            </Text>
            <Icon
              allowFontScaling={false}
              color="#0084F8"
              name="info"
              size={22}
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#0084F8",
                      "fontSize": 22,
                    },
                    undefined,
                    Object {
                      "fontFamily": "icomoon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#0084F8",
                        "fontSize": 22,
                      },
                      undefined,
                      Object {
                        "fontFamily": "icomoon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </Text>
            </Icon>
          </View>
        </View>
      </View>
    </View>
  </CardBody>
</withI18nextTranslation(CardBody)>
`;
